### ISSUE TITLE
[RTK-Query]: `useQuery` hook does not refetch after `resetApiState` (2)

### ISSUE THREAD

**Issue by manceauJb** [reduxjs/redux-toolkit#3778]
### Description:
This issue appears to be a bug in RTK-Query, which should have been resolved by #1735 and #3333. However, it appears that the problem persists.

### Reproduction Steps:
1. Access the following example showcasing the issue: [CodeSandbox](https://codesandbox.io/s/rtkq-resetapistate-query-2mxs7f)
2. Click on the "Rerender Component" button.
3. Click on the "Reset Api State" button.

### Expected Behavior:
After clicking "Reset API State," the request should not remain stuck in a pending status. Instead, it should automatically re-fetch the query. Currently, the request remains pending until the component is re-rendered. Even if we attempt to manually refetch, the query is executed, but the result remains pending.

**Comment by phryneas** [comment]
Yup, I can see that. Weird that it only happens after a rerender 🤔

**Comment by etozhealkhipce** [comment]
@phryneas 

Is there any fixes for this?
I have same issue on my project

**Comment by bennyszantar** [comment]
Same thing in my project.

**Comment by zigcccc** [comment]
any updates re this?
